### PR TITLE
Make all tests async

### PR DESCRIPTION
**Pull request by adam-fowler**

### CODE REVIEW

File: Tests/SotoCoreTests/AWSClientTests.swift [hunk 1]
```diff
@@ -401,14 +399,14 @@ class AWSClientTests: XCTestCase {
             let threadPool = NIOThreadPool(numberOfThreads: 3)
             threadPool.start()
             let fileIO = NonBlockingFileIO(threadPool: threadPool)
-            let fileHandle = try fileIO.openFile(path: filename, mode: .read, eventLoop: httpClient.eventLoopGroup.next()).wait()
+            let fileHandle = try await fileIO.openFile(path: filename, mode: .read, eventLoop: httpClient.eventLoopGroup.next()).get()
```
Comment (0xTim) [inline Tests/SotoCoreTests/AWSClientTests.swift:402]: We could really do with a new async version of FileIO, preferably with an AsyncSequence

File: Tests/SotoCoreTests/Credential/RotatingCredentialProviderTests.swift [hunk 2]
```diff
@@ -27,22 +27,20 @@ import SotoTestUtils
...
 
         init(_ callback: @escaping TestCallback) {
             self.callback = callback
         }
 
-        func getCredential(on eventLoop: EventLoop, logger: Logger) -> EventLoopFuture<Credential> {
-            eventLoop.flatSubmit {
-                return self.callback(eventLoop).map { $0 }
-            }
+        func getCredential(on eventLoop: EventLoop, logger: Logger) async throws -> Credential {
```
Comment (0xTim) [inline Tests/SotoCoreTests/Credential/RotatingCredentialProviderTests.swift:38]: I'm assuming passing the event loop will disappear in a future API

File: Tests/SotoCoreTests/EndpointDiscoveryTests.swift [hunk 3]
```diff
@@ -18,7 +18,8 @@ import SotoCore
 import SotoTestUtils
 import XCTest
 
-class EndpointDiscoveryTests: XCTestCase {
+@available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
```
Comment (0xTim) [inline Tests/SotoCoreTests/EndpointDiscoveryTests.swift:21]: Do we need this in Soto 7?

File: Tests/SotoCoreTests/PaginateTests.swift [hunk 4]
```diff
@@ -13,13 +13,15 @@
...
 import AsyncHTTPClient
+import Atomics
 import NIOCore
 import NIOPosix
 @testable import SotoCore
 import SotoTestUtils
 import XCTest
 
-class PaginateTests: XCTestCase {
+@available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
```
Comment (0xTim) [inline Tests/SotoCoreTests/PaginateTests.swift:23]: As above

**Review by 0xTim — APPROVED** [review]
Some comments but overall good

File: Tests/SotoCoreTests/EndpointDiscoveryTests.swift (hunk 3 above)
Comment (adam-fowler) [reply Tests/SotoCoreTests/EndpointDiscoveryTests.swift:21]: The next PR will fix these

**Pull request merged**